### PR TITLE
[media-library] Refactor to Kotlin 3 - Album migration

### PR DESCRIPTION
**Pull request by barthap**
# Why, How etc...

See #14562 description for details.

---

This is part of a PR stack:

1. #14562
2. #14563
3. **This PR** - rewritten album migration to Kotlin, extracted it to `expo.modules.medialibrary.albums.migration` package 
4. #14565 
5. #14566 
6. Coroutines + refactor error handling (No PR yet)

### CODE REVIEW

File: packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/albums/migration/CheckIfAlbumShouldBeMigrated.kt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    null
+  ).use { albumCursor ->
+    if (albumCursor != null && albumCursor.moveToNext()) {
+      val dataColumnIndex = albumCursor.getColumnIndex(MediaStore.Images.Media.DATA)
+      val fileInAlbum = File(albumCursor.getString(dataColumnIndex))
+      if (fileInAlbum.isFile) {
+        return File(fileInAlbum.parent ?: return null)
+      }
+    }
+  }
```
Comment (lukmccall) [inline packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/albums/migration/CheckIfAlbumShouldBeMigrated.kt:52]:
```suggestion
  )?.use { albumCursor ->
    if (albumCursor.moveToNext()) {
      val dataColumnIndex = albumCursor.getColumnIndex(MediaStore.Images.Media.DATA)
      val fileInAlbum = File(albumCursor.getString(dataColumnIndex))
      if (fileInAlbum.isFile) {
        return File(fileInAlbum.parent ?: return null)
      }
    }
  }
```
You can remove the null check, but I'm not sure if that looks better 🤷‍♂️

File: packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/albums/migration/MigrateAlbum.kt [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import java.io.File
+
+@RequiresApi(Build.VERSION_CODES.R)
+internal class MigrateAlbum(
+  private val context: Context,
+  private val assetFiles: List<AssetFile>,
+  private val albumDirName: String,
+  private val promise: Promise
+) : AsyncTask<Void?, Void?, Void?>() {
+  override fun doInBackground(vararg voids: Void?): Void? {
```
Comment (lukmccall) [inline packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/albums/migration/MigrateAlbum.kt:23]: You can do something with that type to make it looks better. If you use a different type here (Unit or Any), you don't have to return null. However, I know, you're going to rewrite that using corutines. So leave it as it's.

**Review by lukmccall — APPROVED** [review]
Great job 🥇

**Review by kkafar — COMMENTED** [review]
Looks good, no objections

**Pull request merged**